rct229/rules/section5/section5rule36.py

```diff
@@ -0,0 +1,161 @@
+                surface_b = context_item.baseline
+                return (
+                    get_opaque_surface_type(surface_b) == OST.ROOF
+                    and scc[surface_b["id"]] != SCC.UNREGULATED
+                )
+
+            class RoofRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule36.BuildingRule.BuildingSegmentRule.RoofRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    total_skylight_area_b = data["total_skylight_area_b"]
+                    total_skylight_area_p = data["total_skylight_area_p"]
+
+                    roof_b = context.baseline
+                    roof_p = context.proposed
+
+                    total_skylight_area_surface_b = sum(
+                        [
+                            subsurface.get("glazed_area", ZERO.AREA)
+                            + subsurface.get("opaque_area", ZERO.AREA)
+                            for subsurface in find_all("subsurfaces[*]", roof_b)
+                        ],
+                        ZERO.AREA,
+                    )
+                    total_skylight_area_surface_p = sum(
+                        [
+                            subsurface.get("glazed_area", ZERO.AREA)
+                            + subsurface.get("opaque_area", ZERO.AREA)
+                            for subsurface in find_all("subsurfaces[*]", roof_p)
+                        ],
+                        ZERO.AREA,
+                    )
+
+                    return {
+                        "total_skylight_area_b": total_skylight_area_b,
+                        "total_skylight_area_p": total_skylight_area_p,
+                        "total_skylight_area_surface_b": total_skylight_area_surface_b,
+                        "total_skylight_area_surface_p": total_skylight_area_surface_p,
+                    }
+
+                def rule_check(self, context, calc_vals=None, data=None):
+
+                    total_skylight_area_b = calc_vals["total_skylight_area_b"]
+                    total_skylight_area_p = calc_vals["total_skylight_area_p"]
+                    total_skylight_area_surface_b = calc_vals[
+                        "total_skylight_area_surface_b"
+                    ]
+                    total_skylight_area_surface_p = calc_vals[
+                        "total_skylight_area_surface_p"
+                    ]
+
+                    return (
+                        # both segments have no skylight area
+                        total_skylight_area_b == 0 and total_skylight_area_p == 0
+                    ) or (
+                        # both segments have skylight area and ratios are the same
+                        total_skylight_area_b * total_skylight_area_p > 0
```

Comment:
Nice way to articulate that neither can be zero!
It might be good to add a comment explain that that is the purpose of looking at the product.

Reply:
added